internal/resources/provisioner/resource_provisioner.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package provisioner
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	provisioner "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/provisioner"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+type (
+	contextMethodKey struct{}
+)
+
+func ResourceProvisioner() *schema.Resource {
+	return &schema.Resource{
+		Schema:        provisionerSchema,
+		ReadContext:   dataSourceProvisionerRead,
+		CreateContext: resourceProvisionerCreate,
+		UpdateContext: resourceProvisionerInPlaceUpdate,
+		DeleteContext: resourceProvisionerDelete,
+		Description:   "Tanzu Mission Control Provisioner Resource",
+	}
+}
+
+var provisionerSchema = map[string]*schema.Schema{
+	NameKey: {
+		Type:        schema.TypeString,
+		Description: "Name of the provisioner",
+		Required:    true,
+		ForceNew:    true,
+	},
+	ManagementClusterNameKey: {
+		Type:        schema.TypeString,
+		Description: "Name of the management cluster",
```

Comment:
here I feel its better to mention that it is currently not supported for TKG. @vmw-vjn any thoughts?

Reply:
First, I think any mention of applicability of resource - should in the docs (i.e.) docs/resources folder as md file. And not in code-comments in go file. Second - if we need to mention in code comments here, I'd suggest stating that TKG create & use default provisioners. As stated by LCM team in this Slack conversation: https://vmware.slack.com/archives/C0299HW9GH5/p1702905733958259

Reply:
Agree with @vmw-vjn, we need to capture all these details in the document and not in the implementation side

Reply:
@vmw-vjn  let me add more details.

1. These are node go code comments, instead it is the schema description which does appear in the provider documentation as a description of each of the field in the schema which user would refer to while constructing the resource definition. You can see an example in the repo [here](https://registry.terraform.io/providers/vmware/tanzu-mission-control/latest/docs/resources/integration#integration_name)
2. I agree that first of all this limitation should be stated in the TMC doc which we can easily refer to in TMC terraform provider docs. But unfortunately it is missing from there so somewhere in terraform provider doc it should be added to ensure we avoid more questions around the same when user tries to create provisioner on TKG management clusters.

Reply:
There is a bug in the `tfdocplugins` as the description of the schema is not included in the doc generation. Anyways, adding clear description in the schema. 
We definitely should have this captured in doc as well